### PR TITLE
Enable docxImage to pull images from a URL during post-render

### PR DESCRIPTION
**Pull request by sebastian-schlecht**
Enable a second parameter to docxImage called `url` which can be used as an alternative to `src`. In case `url` is set, images are fetched during post-render and then rendered into the document. Files are fetched in parallel. This allows to only download those images which are actually used within the document.

Addresses #3.

### CODE REVIEW

**Comment by sebastian-schlecht** [comment]
Somehow `./node_modules/standard --fix` doesn't fix all the formatting issues, hence the large diff.

File: static/helpers.js [hunk 1]
```diff
@@ -15,36 +15,62 @@
...
-    return new Handlebars.SafeString(`<docxStyle id="${options.hash.id}" textColor="${options.hash.textColor}" />`)
+    return new Handlebars.SafeString(
+      `<docxStyle id="${options.hash.id}" textColor="${options.hash.textColor}" />`
+    )
   }
 
   global.docxImage = function (options) {
-    if (!options.hash.src) {
-      throw new Error('docxImage helper requires src parameter to be set')
+    if (!options.hash.src && !options.hash.url) {
```
Comment (pofider) [inline static/helpers.js]: We would prefer having just `src` param and don't use extra  `url` param. The same as the `img` html tag has.
You can pass to the src data uri like `src='data:image/png;base64,...'` but also `src='https://foo.com/img'`. You would just need to deal with recognizing what is data uri and what you should request.

**Comment by sebastian-schlecht** [comment]
Thanks for the comments & feedback. Will probably implement the changes tomorrow.

File: static/helpers.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* eslint no-unused-vars: 1 */
 /* eslint no-new-func: 0 */
 /* *global __rootDirectory */
-;(function (global) {
```
[the remark this answers is not in the document]
Comment (sebastian-schlecht) [reply static/helpers.js]: Got autoformatted unfortunately

File: static/helpers.js (hunk 1 above)
Comment (sebastian-schlecht) [reply static/helpers.js]: good idea, makes sense

**Comment by sebastian-schlecht** [comment]
@pofider I implemented your feedback on the PR. I can fix the remaining formatting diff if you want, but I would need a style-config for that.

**Pull request merged**